llvm/lib/Analysis/ValueTracking.cpp

```diff
@@ -5805,6 +5805,36 @@ void computeKnownFPClass(const Value *V, const APInt &DemandedElts,
 
     break;
   }
+  case Instruction::BitCast: {
+    const Type *Ty = Op->getType();
+    const Value *Casted = Op->getOperand(0);
+    if (Ty->isVectorTy() || !Casted->getType()->isIntOrIntVectorTy())
```

Comment:
Why not handle `vector -> vector` casts? You can use `m_ElementWiseBitCast` to rule out `vector <-> scalar` casts.

Reply:
Good point, added.